src/http/sr-modules-operators-keys/sr-modules-operators-keys.e2e-spec.ts

```diff
@@ -0,0 +1,254 @@
+        await moduleStorageService.upsert(curatedModule, 1);
+      });
+
+      afterAll(async () => {
+        await cleanDB();
+      });
+
+      it('should return all keys for request without filters', async () => {
+        // Get all keys and operators without filters
+        const resp = await request(app.getHttpServer()).get(`/v1/modules/${dvtModule.id}/operators/keys`);
+
+        expect(resp.status).toEqual(200);
+        expect(resp.body.data.operators).toEqual(expect.arrayContaining([operatorOneDvt, operatorTwoDvt]));
+        expect(resp.body.data.keys).toEqual(expect.arrayContaining(dvtModuleKeys));
+        expect(resp.body.data.module).toEqual(dvtModuleResp);
+        expect(resp.body.meta).toEqual({
+          elBlockSnapshot: {
+            blockNumber: elMeta.number,
+            blockHash: elMeta.hash,
+            timestamp: elMeta.timestamp,
+          },
+        });
+      });
+
+      it('should return 400 error if operatorIndex is not a number', async () => {
+        const resp = await request(app.getHttpServer())
+          .get(`/v1/modules/${dvtModule.id}/operators/keys`)
+          .query({ used: false, operatorIndex: 'one' });
+        expect(resp.status).toEqual(400);
+        expect(resp.body).toEqual({
+          error: 'Bad Request',
+          message: ['operatorIndex must not be less than 0', 'operatorIndex must be an integer number'],
+          statusCode: 400,
+        });
+      });
+
+      it('should return 400 error if used is not a boolean value', async () => {
+        const resp = await request(app.getHttpServer())
+          .get(`/v1/modules/${dvtModule.id}/operators/keys`)
+          .query({ used: 0, operatorIndex: 2 });
+        expect(resp.status).toEqual(400);
+        expect(resp.body).toEqual({ error: 'Bad Request', message: ['used must be a boolean value'], statusCode: 400 });
+      });
+
+      it('should return used keys and operator one', async () => {
+        const resp = await request(app.getHttpServer())
+          .get(`/v1/modules/${dvtModule.id}/operators/keys`)
+          .query({ used: true, operatorIndex: 1 });
+
+        const expectedKeys = dvtModuleKeys.filter((key) => key.used && key.operatorIndex == 1);
+
+        expect(resp.status).toEqual(200);
+        expect(resp.body.data.operators).toEqual(expect.arrayContaining([operatorOneDvt]));
+        expect(resp.body.data.keys).toEqual(expect.arrayContaining(expectedKeys));
+        expect(resp.body.data.module).toEqual(dvtModuleResp);
+        expect(resp.body.meta).toEqual({
+          elBlockSnapshot: {
+            blockNumber: elMeta.number,
+            blockHash: elMeta.hash,
+            timestamp: elMeta.timestamp,
+          },
+        });
+      });
+
+      it('should return unused keys and operator one', async () => {
+        const resp = await request(app.getHttpServer())
+          .get(`/v1/modules/${dvtModule.id}/operators/keys`)
+          .query({ used: false, operatorIndex: 1 });
```

Comment:
It would be nice to also have a test for operatorIndex = 0, since a value of 0 may pass some checks differently

Reply:
Do you mean to check whether setting operatorIndex to 0 will result in no validation error (400)?

Reply:
Found mistake in code, thank you 
wrong filter